Shared/messages/MessageFeed.cpp

```diff
@@ -145,9 +230,86 @@ QUrl MessageFeed::thumbnailUrl() const
 /*!
   \brief Sets the (local file) URL of the thumbnail to use for this feed to \a thumbnailUrl.
  */
-void MessageFeed::setThumbnailUrl(const QUrl& thumbnailUrl)
+void MessageFeed::setThumbnailUrl(const QUrl &thumbnailUrl)
 {
   m_thumbnailUrl = thumbnailUrl;
 }
 
+/*!
+  \brief Adds the \l Message \a message to the overlay. Returns whether adding was successful.
+ */
+bool MessageFeed::addMessage(const Message &message)
+{
+  static QString additionalErrorMessage = "DSA - MessageFeed";
+  const auto messageId = message.messageId();
+  if (messageId.isEmpty())
+  {
+    emit errorOccurred(Error("Failed to add message - message ID is empty", additionalErrorMessage, ExtendedErrorType::None));
+    return false;
+  }
+
+  if (message.messageType() != this->feedMessageType())
+  {
+    emit errorOccurred(Error("Failed to add message - message type mismatch", additionalErrorMessage, ExtendedErrorType::None));
+    return false;
+  }
+
+  const auto symbolId = message.symbolId();
+  const auto geometry = message.geometry();
+  const auto messageAction = message.messageAction();
+
+  switch (messageAction)
+  {
+  case Message::MessageAction::Remove:
+    this->deleteEntityAsync(messageId).then([]{});
```

Comment:
Is any .then required here? If there's nothing to do, is it enough to just call it? If you get a warning because of unused return type, you can do this:

```suggestion
    auto future = this->deleteEntityAsync(messageId);
    Q_UNUSED(future)
```

That would better convey the intent here IMO.

Reply:
Correct. No then necessary. Applied the suggested fix.